### PR TITLE
 Remove `safeStrtol` from conversion.h (use `tryTo<long>` instead)

### PR DESCRIPTION
**Pull request by akindyakov**
One more step to use `tryTo<long>` instead of `safeStrto*`

Based on PR #4767

### CODE REVIEW

**Comment by akindyakov** [comment]
@guliashvili , no, I don't think that explicit specification of the conversion base does make much sense. But for some reason authors of the changed code think in a different way. Maybe they think that it can help to read or something else. My point was just to keep the code close to original and make my change smaller.

Thank you for approval!

File: osquery/config/packs.cpp [hunk 1]
```diff
@@ -74,9 +74,9 @@ size_t getMachineShard(const std::string& hostname = "", bool force = false) {
   auto hn_hash = getBufferSHA1(hn.c_str(), hn.size());
 
   if (hn_hash.size() >= 2) {
-    long hn_char;
-    if (safeStrtol(hn_hash.substr(0, 2), 16, hn_char)) {
-      shard = (hn_char * 100) / 255;
+    auto const hn_char = tryTo<long>(hn_hash.substr(0, 2), 16);
```
Comment (obelisk) [inline osquery/config/packs.cpp]: This variable contains the type name `char` but it is `tryTo<long>`

File: osquery/config/packs.cpp [hunk 2]
```diff
@@ -74,9 +74,9 @@ size_t getMachineShard(const std::string& hostname = "", bool force = false) {
...
+    if (hn_char) {
```
Comment (obelisk) [inline osquery/config/packs.cpp]: Prefer explicit checks.

File: osquery/config/packs.cpp [hunk 3]
```diff
@@ -90,12 +90,13 @@ size_t restoreSplayedValue(const std::string& name, size_t interval) {
     // This query name existed before, check the last requested interval.
     auto details = osquery::split(content, ":");
     if (details.size() == 2) {
-      long last_interval, last_splay;
-      if (safeStrtol(details[0], 10, last_interval) &&
-          safeStrtol(details[1], 10, last_splay)) {
-        if (last_interval == static_cast<long>(interval) && last_splay > 0) {
+      auto const last_interval = tryTo<long>(details[0], 10);
+      auto const last_splay = tryTo<long>(details[1], 10);
+      if (last_interval && last_splay) {
```
Comment (obelisk) [inline osquery/config/packs.cpp]: Prefer explicit checks

File: osquery/core/darwin/iokit.hpp [hunk 4]
```diff
@@ -44,10 +44,11 @@ long long int getNumIOKitProperty(const CFMutableDictionaryRef& details,
                                   const std::string& key);
 
 inline void idToHex(std::string& id) {
-  long base = 0;
-  if (safeStrtol(id, 10, base)) {
+  auto base = tryTo<long>(id, 10);
+  if (base) {
```
Comment (obelisk) [inline osquery/core/darwin/iokit.hpp]: Prefer explicit checks

File: osquery/core/system.cpp [hunk 5]
```diff
@@ -426,11 +426,15 @@ bool PlatformProcess::cleanup() const {
 #ifndef WIN32
 
 static inline bool ownerFromResult(const Row& row, long& uid, long& gid) {
-  if (!safeStrtol(row.at("uid"), 10, uid) ||
-      !safeStrtol(row.at("gid"), 10, gid)) {
-    return false;
+  auto const uid_exp = tryTo<long>(row.at("uid"), 10);
+  auto const gid_exp = tryTo<long>(row.at("gid"), 10);
+  if (uid_exp) {
```
Comment (obelisk) [inline osquery/core/system.cpp]: Prefer explicit checks

File: osquery/core/system.cpp [hunk 6]
```diff
@@ -426,11 +426,15 @@ bool PlatformProcess::cleanup() const {
...
+    uid = uid_exp.get();
   }
-  return true;
+  if (gid_exp) {
```
Comment (obelisk) [inline osquery/core/system.cpp]: Prefer explicit checks

File: osquery/tables/system/linux/memory_info.cpp [hunk 7]
```diff
@@ -49,9 +49,9 @@ QueryData getMemoryInfo(QueryContext& context) {
       // Look for mapping
       for (const auto& singleMap : kMemInfoMap) {
         if (line.find(singleMap.second) == 0) {
-          long value = 0;
-          if (safeStrtol(tokens[1], 10, value)) {
-            r[singleMap.first] = BIGINT(value * 1024l);
+          auto const value_exp = tryTo<long>(tokens[1], 10);
+          if (value_exp) {
```
Comment (obelisk) [inline osquery/tables/system/linux/memory_info.cpp]: Prefer explicit checks

File: osquery/tables/system/linux/user_groups.cpp [hunk 8]
```diff
@@ -23,8 +23,8 @@ QueryData genUserGroups(QueryContext& context) {
   if (context.constraints["uid"].exists(EQUALS)) {
     std::set<std::string> uids = context.constraints["uid"].getAll(EQUALS);
     for (const auto& uid : uids) {
-      long auid{0};
-      if (safeStrtol(uid, 10, auid) && (pwd = getpwuid(auid)) != nullptr) {
+      auto const auid_exp = tryTo<long>(uid, 10);
+      if (auid_exp && (pwd = getpwuid(auid_exp.get())) != nullptr) {
```
Comment (obelisk) [inline osquery/tables/system/linux/user_groups.cpp]: Prefer explicit checks

File: osquery/tables/system/linux/users.cpp [hunk 9]
```diff
@@ -54,10 +54,10 @@ QueryData genUsers(QueryContext& context) {
   if (context.constraints["uid"].exists(EQUALS)) {
     auto uids = context.constraints["uid"].getAll(EQUALS);
     for (const auto& uid : uids) {
-      long auid{0};
-      if (safeStrtol(uid, 10, auid)) {
+      auto const auid_exp = tryTo<long>(uid, 10);
+      if (auid_exp) {
```
Comment (obelisk) [inline osquery/tables/system/linux/users.cpp]: Prefer explicit checks

File: osquery/config/packs.cpp (hunk 1 above)
Comment (akindyakov) [reply osquery/config/packs.cpp]: Fixed name `hn_char` to `hn_num`, thanks!

**Comment by akindyakov** [comment]
Thank you for review, @obelisk ! I've fixed up, what you mention. Also, I am going to add one more validation method `Expected::isValue` to make checks if expected is valid less verbose.
Smth like:
```c++
if (exp.isValue()) {
```
instead of:
```c++
if (!exp.isError()) {
```

**Comment by guliashvili** [comment]
@obelisk ping

**Pull request merged**